multierror.go

```diff
@@ -49,3 +50,7 @@ func (e *Error) GoString() string {
 func (e *Error) WrappedErrors() []error {
 	return e.Errors
 }
+
+func (e *Error) MarshalJSON() ([]byte, error) {
+	return json.Marshal(e.Errors)
```

Comment:
ErrorWithAttrs will be. Also we do marshal errors in our logging package. While this will give us no useable marshaled info in most cases, the alternative is 1) logging package getting upset, or 2) changing the logging package to not marshal errors at all.